tools/ci_build/github/apple/test_apple_packages.py

```diff
@@ -112,7 +112,9 @@ def _test_apple_packages(args):
         subprocess.run(["pod", "cache", "clean", "--all"], shell=False, check=True, cwd=target_proj_path)
 
         # install pods
-        subprocess.run(["pod", "install"], shell=False, check=True, cwd=target_proj_path)
+        # set env to skip macos test targets accordingly
+        os.environ['SKIP_MACOS_TEST'] = "true" if args.skip_macos_test else "false"
+        subprocess.run(["pod", "install"], shell=False, check=True, cwd=target_proj_path, env=os.environ)
```

Comment:
let's just modify a copy of os.environ instead of updating it.
```suggestion
        env = os.environ.copy()
        env['SKIP_MACOS_TEST'] = "true" if args.skip_macos_test else "false"
        subprocess.run(["pod", "install"], shell=False, check=True, cwd=target_proj_path, env=env)
```